ServerCore/Pages/PuzzleApiController.cs

```diff
@@ -1,65 +1,97 @@
+        private readonly UserManager<IdentityUser> userManager;
+
+        public PuzzleApiController(PuzzleServerContext context, UserManager<IdentityUser> userManager)
+        {
             this.context = context;
-            this.userManager = userManager;
-        }
-
-        /// <summary>
-        /// Allows external puzzles to validate a team password
-        /// </summary>
-        /// <param name="teamPassword">Potential team password</param>
-        /// <returns>True if the password belongs to a team, false otherwise</returns>
-        [HttpGet]
-        [Route("api/puzzleapi/validateteampassword")]
-        public async Task<ActionResult<bool>> GetValidateTeamPasswordAsync(string teamPassword)
-        {
-            if (string.IsNullOrWhiteSpace(teamPassword))
-            {
-                return false;
-            }
-
-            return await (from team in context.Teams
-                          where team.Password == teamPassword
-                          select team).AnyAsync();
-        }
-
+            this.userManager = userManager;
+        }
+
+        /// <summary>
+        /// Allows external puzzles to validate a team password
+        /// </summary>
+        /// <param name="teamPassword">Potential team password</param>
+        /// <returns>True if the password belongs to a team, false otherwise</returns>
+        [HttpGet]
+        [Route("api/puzzleapi/validateteampassword")]
+        public async Task<ActionResult<bool>> GetValidateTeamPasswordAsync(string teamPassword)
+        {
+            if (string.IsNullOrWhiteSpace(teamPassword))
+            {
+                return false;
+            }
+
+            return await (from team in context.Teams
+                          where team.Password == teamPassword
+                          select team).AnyAsync();
+        }
+
+        /// <summary>
+        /// Allows external puzzles to validate a team password
+        /// </summary>
+        /// <param name="teamPassword">Potential team password</param>
+        /// <returns>True if the password belongs to a team, false otherwise</returns>
+        [HttpGet]
+        [Route("api/puzzleapi/getmailinfo")]
+        public async Task<ActionResult<MailInfo>> GetMailInfoAsync(string eventId, int puzzleId)
+        {
+            Event currentEvent = await EventHelper.GetEventFromEventId(context, eventId);
+            Team team = await UserEventHelper.GetTeamForCurrentPlayer(context, currentEvent, User, userManager);
+
+            MailInfo mailInfo = new MailInfo();
+
+            mailInfo.PuzzleName = await (from puzzle in context.Puzzles
+                                         where puzzle.ID == puzzleId
+                                         select puzzle.Name).FirstOrDefaultAsync();
+            mailInfo.TeamName = team.Name;
+
+            // replace commas with semicolons for better email support
+            mailInfo.TeamContactEmail = team.PrimaryContactEmail.Replace(',', ';');

```

Comment:
Does this need a null check? I thought this was optional

Reply:
Probably, just added one